scipy/optimize/_trustregion_exact.py

```diff
@@ -0,0 +1,425 @@
+        # "Matrix computations". Forth Edition. JHU press., p.165.
+        self.CLOSE_TO_ZERO = self.dimension*self.EPS*self.hess_inf
+
+    def _initial_values(self, trust_radius):
+        """Given a trust radius, return a good initial guess for
+        the damping factor, the lower bound and the upper bound.
+        The values were chosen accordingly to the guidelines on
+        section 7.3.8 (p. 192) from [1].
+        """
+
+        # Upper bound
+        ub = max(0, self.jac_mag/trust_radius+min(-self.hess_gershgorin_lb,
+                                                  self.hess_fro,
+                                                  self.hess_inf))
+
+        # Lower bound
+        lb = max(0, -min(self.hess.diagonal()),
+                 self.jac_mag/trust_radius-min(self.hess_gershgorin_ub,
+                                               self.hess_fro,
+                                               self.hess_inf))
+
+        # Improve bounds with previous info
+        if trust_radius < self.previous_trust_radius:
+            lb = max(self.lb, lb)
+
+        # Damping factor
+        if lb == 0:
+            damp = 0
+        else:
+            damp = max(np.sqrt(lb*ub), lb+self.UPDATE_COEF*(ub-lb))
+
+        return damp, lb, ub
+
+    def solve(self, trust_radius):
+        """Solve quadratic subproblem"""
+
+        damp, lb, ub = self._initial_values(trust_radius)
+        n = self.dimension
+        hits_boundary = True
+        already_factorized = False
+        self.niter = 0
+
+        while True:
+
+            # Compute Cholesky factorization
+            if already_factorized:
+                already_factorized = False
+            else:
+                H = self.hess+damp*np.eye(n)
+                U, info = self.cholesky(H, lower=False,
+                                        overwrite_a=False,
+                                        clean=True)
+
+            # Count iteration
+            self.niter += 1
+
+            # Check if factorization succeded
+            if info == 0 and self.jac_mag > self.CLOSE_TO_ZERO:
+                # Successfull factorization
+
+                # Solve `U.T s = -g`
+                s = solve_triangular(U, -1*self.jac, trans='T')
+
+                # Solve `U p = s`
+                p = solve_triangular(U, s, trans='N')
+
+                # Compute `p` norm
+                pnorm = norm(p)
+
+                # Check for interior convergence
+                if pnorm <= trust_radius and damp == 0:
+                    hits_boundary = False
+                    break 
+
+                # Solve `U.T w = p`
+                w = solve_triangular(U, p, trans='T')
+
+                # Compute `w` norm
+                wnorm = norm(w)
+
+                # Compute Newton step accordingly to
+                # formula (4.44) p.87 from ref [2]
+                delta_damp = (pnorm/wnorm)**2*(pnorm-trust_radius)/trust_radius
+                damp_next = damp + delta_damp
```

Comment:
Ok I changed it to ``lambda_new`` (to be coherent with the changes on the other variables)